template/web/wp-content/themes/<%= slug %>/app/Repositories/PostRepository.php

```diff
@@ -7,10 +7,24 @@
 
 namespace Studiometa\Repositories;
 
+use Timber\PostQuery;
+
 /** Class */
-class PostTypeRepository extends Repository {
+class PostRepository extends Repository {
+	const CLASS_TYPE = '\Timber\Post';
 	const POST_TYPES = array( 'post' ); // Main post types.
 
+	/**
+	 * Implement do_query
+	 *
+	 * @param array  $params for the query.
+	 * @param string $class_type to return.
+	 * @return \Timber\PostQuery
+	 */
+	public function do_query( $params, $class_type ) {
+		return new PostQuery( $params, self::CLASS_TYPE );
```

Comment:
If you use `self::CLASS_TYPE` and create a class who inherit from this class it will use the parent constant, not the child one. So you should use `static::CLASS_TYPE` instead.

See https://stackoverflow.com/a/13613718